src/beast/beast/hash/hash_append.h

```diff
@@ -173,454 +213,256 @@ static_assert (is_contiguously_hashable <
 
 template <class Hasher, class T>
 inline
-typename std::enable_if
+std::enable_if_t
 <
-    is_contiguously_hashable<T>::value
->::type
-hash_append (Hasher& h, T const& t) noexcept
+    is_contiguously_hashable<T, Hasher>::value
+>
+hash_append(Hasher& h, T const& t) noexcept
 {
-    h.append (&t, sizeof(t));
+    h(std::addressof(t), sizeof(t));
 }
 
 template <class Hasher, class T>
 inline
-typename std::enable_if
-<
-    std::is_floating_point<T>::value
->::type
-hash_append (Hasher& h, T t) noexcept
-{
-    // hash both signed zeroes identically
-    if (t == 0)
-        t = 0;
-    h.append (&t, sizeof(t));
-}
-
-// arrays
-
-template <class Hasher, class T, std::size_t N>
-inline
-typename std::enable_if
+std::enable_if_t
 <
-    !is_contiguously_hashable<T>::value
->::type
-hash_append (Hasher& h, T (&a)[N]) noexcept
+    !is_contiguously_hashable<T, Hasher>::value &&
+    (std::is_integral<T>::value || std::is_pointer<T>::value || std::is_enum<T>::value)
```

Comment:
Nope.  This hash_append should only be used for built-in integers, pointers or enums.  Other types (such as `base_uint`) may declare themselves as `uniquely_represented`, but should not pick up this overload.